### PR TITLE
Add option to ask peer for WINS address

### PR DESCRIPTION
**Pull request by mjeveritt**
Signed-off-by: Michael Everitt <gentoo@veremit.xyz>
Signed-off-by: Lars Wendler <polynomial-c@gentoo.org>

### CODE REVIEW

**Comment by mjeveritt** [comment]
First of a patch series from Gentoo Linux. Test PR to get format right !

**Comment by mjeveritt** [comment]
Hopefully we can pull some of these into 2.4.10 vis #245 :smile:

**Comment by Neustradamus** [comment]
@paulusmack: What do you think?

**Comment by paulusmack** [comment]
I think there is a problem with the patch. As the code currently stands (without the patch), if we initially don't ask for MS_WINS[12] in our configure-request, and the peer NAKs with a request with MS_WINS[12] in it, then our next configure-request will include MS_WINS[12]. You seem to have broken that logic, because we now need go->req_wins[12] set to 1 to include MS_WINS[12] in our configure-request, and the logic at the end of ipcp_nakci that handles the case where the peer is NAKing with things we didn't ask for (because it wants us to ask for them) hasn't been modified to set try.req_wins[12].

In other words the code for the CI_MS_WINS[12] cases near the end of ipcp_nakci() needs to be modified to look more similar to the code for the CI_MS_DNS[12] cases.

**Comment by Neustradamus** [comment]
@mjeveritt: Have you looked the comment of @paulusmack?

**Comment by mjeveritt** [comment]
> I think there is a problem with the patch. As the code currently stands (without the patch), if we initially don't ask for MS_WINS[12] in our configure-request, and the peer NAKs with a request with MS_WINS[12] in it, then our next configure-request will include MS_WINS[12]. You seem to have broken that logic, because we now need go->req_wins[12] set to 1 to include MS_WINS[12] in our configure-request, and the logic at the end of ipcp_nakci that handles the case where the peer is NAKing with things we didn't ask for (because it wants us to ask for them) hasn't been modified to set try.req_wins[12].
> 
> In other words the code for the CI_MS_WINS[12] cases near the end of ipcp_nakci() needs to be modified to look more similar to the code for the CI_MS_DNS[12] cases.

ACK, ok will look more deeply into this. Haven't yet found the applicable code, but hope to spend some time before end-of-month.

> @mjeveritt: Have you looked the comment of @paulusmack?

See above :)

**Comment by Neustradamus** [comment]
@mjeveritt: Have you found the time?

**Comment by mjeveritt** [comment]
> @mjeveritt: Have you found the time?

@Neustradamus : Thanks for the ping, aiming to allocate some time this week on this.

**Comment by mjeveritt** [comment]
> In other words the code for the CI_MS_WINS[12] cases near the end of ipcp_nakci() needs to be modified to look more similar to the code for the CI_MS_DNS[12] cases.

@paulusmack : fixing end of the ipcp_nakci() looks fine, do you need me to add `#define NAKCIWINS` etc. at the top, too?

**Comment by paulusmack** [comment]
> @paulusmack : fixing end of the ipcp_nakci() looks fine, do you need me to add `#define NAKCIWINS` etc. at the top, too?

I don't have a strong opinion about that, as long as the result is readable.

**Comment by mjeveritt** [comment]
Hang on a minute Git .. I coulda sworn I committed the code above as noted in https://github.com/ppp-project/ppp/pull/297#issuecomment-891586874 ...

**Comment by mjeveritt** [comment]
> Hang on a minute Git .. I coulda sworn I committed the code above as noted in [#297 (comment)](https://github.com/ppp-project/ppp/pull/297#issuecomment-891586874) ...

My bad, in switching branches locally, I missed that the macros had already been present in the first commit #182f4ca !

**Comment by Neustradamus** [comment]
@paulusmack: What do you think?

**Comment by mjeveritt** [comment]
I can, of course, squash the commits, and fix the log message, FYI; but might as well get the changes right first! :smiley:

**Comment by paulusmack** [comment]
The change looks fine; if you'd like to redo it as a single commit and fix the log message, that would be great, thanks.

**Comment by Neustradamus** [comment]
Thanks to @mjeveritt for the PR and @paulusmack for merging.

**Comment by mjeveritt** [comment]
> Thanks to @mjeveritt for the PR and @paulusmack for merging.

Hurrah - onto the next one! :D (or two, or three..)
